matlab/test/arrow/type/tDisplay.m

```diff
@@ -0,0 +1,133 @@
+                " (" +  float32Link + ", " + timestampLink + ") array with properties:" + newline;
+            body = "    " + "ID";
+            footer = string(newline);
+            expectedDisplay = char(strjoin([header body' footer], newline));
+            actualDisplay = evalc('disp(typeArray)');
+            testCase.verifyDisplay(actualDisplay, expectedDisplay);
+        end
+
+        function NonScalarArraySameTypes(testCase)
+            % Verify the display of a scalar, homogeneous arrow.type.Type array:
+            % 
+            % Example:
+            %
+            %  1×2 heterogeneous FixedWidthType (Float32Type, TimestampType) array with properties:
+            %
+            %    ID
+
+            timestampType1 = arrow.timestamp(TimeZone="Pacific/Fiji");
+            timestampType2 = arrow.timestamp(TimeUnit="Second");
+            typeArray = [timestampType1 timestampType2]';
+
+            timestampLink = makeDisplayLink(FullClassName="arrow.type.TimestampType", ClassName="TimestampType", BoldFont=true);
+            sizeString = makeSizeString(size(typeArray));
+            header = "  " + sizeString + " " + timestampLink + " array with properties:" + newline;
+            body = strjust(["ID"; "TimeUnit"; "TimeZone"], "left");
+            body = "    " + body;
+            footer = string(newline);
+            expectedDisplay = char(strjoin([header body' footer], newline));
+            actualDisplay = evalc('disp(typeArray)');
+            testCase.verifyDisplay(actualDisplay, expectedDisplay);
+        end
+    end
+
+    methods
+        function verifyDisplay(testCase, actualDisplay, expectedDisplay)
+            % When the MATLAB GUI is running, '×' (char(215)) is used the 
+            % delimiter between dimension values. However, when the GUI is
+            % not running, 'x' (char(120)) is used as the delimiter.
+            % To account for this discrepancy, check if actualDisplay 
+            % contains char(215). If not, replace all instances of
+            % char(215) in expectedDisplay with char(120).
+
+            tf = contains(actualDisplay, char(215));
+            if ~tf
+                idx = strfind(expectedDisplay, char(215));
+                expectedDisplay(idx) = char(120);
+            end
+            testCase.verifyEqual(actualDisplay, expectedDisplay);
+        end
+    end
+end
+
+function link = makeDisplayLink(opts)
+    arguments
+        opts.FullClassName(1, 1) string
+        opts.ClassName(1, 1) string
+        opts.BoldFont(1, 1) logical
+    end
+
+    if opts.BoldFont
+        link = compose("<a href=""matlab:helpPopup %s"" style=""font-weight:bold"">%s</a>", ...
+            opts.FullClassName, opts.ClassName);
+    else
+        link = compose("<a href=""matlab:helpPopup %s"">%s</a>", opts.FullClassName, opts.ClassName);
+    end
+end
+
+function sizeString = makeSizeString(arraySize)
```

Comment:
Maybe we should rename this to something like `makeDimensionDisplay`?

Reply:
Agreed. This name isn't great. I decided to rename `makeSizeString` to `makeDimensionString` and `makeDisplayLink` to `makeLinkString`.